care/users/api/viewsets/userskill.py

```diff
@@ -1,25 +1,49 @@
 from django.db.utils import IntegrityError
 from rest_framework.exceptions import ValidationError
 from rest_framework.generics import get_object_or_404
+from rest_framework.permissions import SAFE_METHODS, BasePermission
 from rest_framework.viewsets import GenericViewSet, ModelViewSet
 
+from care.facility.models.base import READ_ONLY_USER_TYPES
 from care.users.api.serializers.userskill import UserSkillSerializer
-from care.users.models import UserSkill
+from care.users.models import User, UserSkill
 from care.utils.queryset.user import get_users
 
 
+class UserSkillPermission(BasePermission):
+    def has_permission(self, request, view):
+        username = view.kwargs.get("users_username")
+        requesting_user = request.user
+        user = get_object_or_404(User, username=username)
+
+        if request.method not in SAFE_METHODS:
+            if requesting_user == user:
+                return True
+
+            if (
+                requesting_user.user_type < User.TYPE_VALUE_MAP["DistrictAdmin"]
+                or requesting_user.user_type in READ_ONLY_USER_TYPES
+            ):
+                return False
+
+            if not requesting_user.user_type >= user.user_type:
```

Comment:
There is no check that confirms if the user is part of the same dist as the dist admin or the same state as the state admin here, but i will approve this PR for now. Do make these changes later or start a new issue for it. 

This should ideally be a helper function that checks if a user has control over another user. this might be useful in many other occasions.